### ISSUE TITLE
Class templating brakes function param annotation

### ISSUE THREAD

**Issue by CristinCebotari** [vimeo/psalm#9790]
The templating of `TValue` adds an [InvalidArgument](https://psalm.dev/004) which makes no sense.

Here's the code with the templating, as you can see it's not used in the function's PHPDoc
https://psalm.dev/r/48b28f27b6

Here's the code without the templating of `TValue` which reports no issues
https://psalm.dev/r/ba18da27b8

**Comment by orklah** [comment]
yeah, strange. Replacing the inheritDoc by the actual type in the parent method solves the issue
